### PR TITLE
completed navbar feature

### LINKED ISSUE #2: 2. As a user, I want to be able to navigate to all the pages in the application by clicking the links in the nav bar
## Summary

Users need to be able to navigate between the pages of our application in order to access all of its features. There is a navigation bar in the `Layout.jsx` component, but it's not quite feature-complete: the links in the nav element don't actually allow the user to navigate from page to page.

## Acceptance criteria
- [x] The `nav` element is added to the `Layout` component
- [x] Links to the "Home", "List", and "Add item" pages of the app are In the `nav` element
- [x] The links all function as expected using  the `NavLink` component from `react-router-dom`

## Notes
- The [official docs for the `NavLink` component](https://reactrouterdotcom.fly.dev/docs/en/v6/components/nav-link) will come in very handy!
- While a standard anchor element (`<a>`) _could_ allow a user to navigate from page to page, the `NavLink` component adds additional behavior that is essential to our app functioning correctly!

### PR DESCRIPTION
**Pull request by hansontram**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

Olivia and I implemented NavLink to the Layout.jsx component to ensure the nav bar is feature-complete & allows the user to navigate thru the site.

## Related Issue

closes #2

## Acceptance Criteria

 The nav element is added to the Layout component
 Links to the "Home", "List", and "Add item" pages of the app are In the nav element
 The links all function as expected using the NavLink component from react-router-dom

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓ | :bug: Bug fix              |
|     | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria
Click thru the nav at the bottom of the page

### CODE REVIEW

File: src/views/Layout.jsx
Comment (01001101CK) [inline src/views/Layout.jsx:1]: Awesome job!! I wonder if we need anchor tag here? For example, 
```
<a href="#" className="Nav-link">
	<NavLink to="/" className="Nav-link">
			Home
</a> 
```
Can be simplified with `<NavLink to="/" className="Nav-link">Home</NavLink>`

Comment (itsoliviasparks) [reply src/views/Layout.jsx:1]: The NavLink replaces the traditional anchor tag, so we wouldn't need to add an a tag in addition.
The code base originally had each link split out to 3 lines, but I'm open to moving each NavLink tag to one line if everyone else is! :)

**Review by 01001101CK — APPROVED** [review]
Awesome job!!!

**Review by lizzypine — APPROVED** [review]
Looks great! Nice and concise.

**Review by vividavila98 — APPROVED** [review]
Great job using react-router!! Go ahead and remove the TODO comment & just for practice, try to include something in the "How to test" section even if it's super simple :)

**Review by laurenyz — APPROVED** [review]
Nice! Short and sweet 🙌

**Pull request merged**